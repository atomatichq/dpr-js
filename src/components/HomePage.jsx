import React from 'react'
import { Link } from 'react-router'

class HomePage extends React.Component {

  constructor(props, context) {
    super(props, context)
  }

  render() {
    return (
      <div>
        <div className="d-flex justify-content-center mb-4">
          <h1>Welcome to Data Package Registry</h1>
        </div>
        <div className="d-flex justify-content-center">
          <Link to={'/core'} className="btn btn-secondary">Core Data Package</Link>
        </div>
      </div>
    )
  }
}
export default HomePage
