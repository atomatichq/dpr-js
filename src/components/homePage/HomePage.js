import React from "react";
import {browserHistory} from "react-router";
import "./HomePage.css";

class HomePage extends React.Component {

  constructor(props, context) {
    super(props, context);
    this._handleSubmit = this._handleSubmit.bind(this);
    this._handleChange = this._handleChange.bind(this);
    this.state = {value: ''};
  }

  _handleSubmit(event){
    event.preventDefault();
    let publisherName = this.state.value;
    browserHistory.push(`/${publisherName}`);
  }
  _handleChange(event) {
    this.setState({value: event.target.value});
  }

  render() {
    return (
      <div className="jumbotron">
        <div className="container">
          <div className="col-md-12">
            <form onSubmit={this._handleSubmit}>
              <div id="custom-search-input" >
                <div className="input-group col-lg-12">
                  <input type="text" className="form-control input-lg" placeholder="Publisher..." value={this.state.value} onChange={this._handleChange} />
                  <span className="input-group-btn">
                    <button className="btn btn-info btn-lg" type="submit">
                        <i className="glyphicon glyphicon-search"/>
                    </button>
                  </span>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}
export default HomePage;

