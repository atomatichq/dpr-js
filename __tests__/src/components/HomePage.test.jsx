import React from 'react'
import renderer from 'react-test-renderer'
import HomePage from '../../../src/components/HomePage'

test('Rendered output matches previously created snapshot', () => {
  const component = renderer.create(<HomePage />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
