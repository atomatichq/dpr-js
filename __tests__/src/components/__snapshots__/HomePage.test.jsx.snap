exports[`test Rendered output matches previously created snapshot 1`] = `
<div>
  <div
    className="d-flex justify-content-center mb-4">
    <h1>
      Welcome to Data Package Registry
    </h1>
  </div>
  <div
    className="d-flex justify-content-center">
    <a
      className="btn btn-secondary"
      onClick={[Function]}
      style={Object {}}>
      Core Data Package
    </a>
  </div>
</div>
`;
